### ISSUE TITLE
Support spdx 2.3

### ISSUE THREAD

**Issue by ankitm123** [anchore/syft#1292]
**What would you like to be added**:

The latest version of spdx specification is 2.3. SBOMs produced by Syft are still on version 2.2.
It would be nice to add support for 2.3. Other sbom authoring tools like ko, apko and kubernetes sbom tool already support producing sbom with version 2.3 as the default.

https://spdx.github.io/spdx-spec/v2.3/

**Comment by kzantow** [comment]
Yes, we definitely want this, too! We are currently using the [spdx/tools-golang](https://github.com/spdx/tools-golang) library for SPDX support and have opened a [PR to get SPDX 2.3 support added there](https://github.com/spdx/tools-golang/pull/164), which seems like it might be close, after which time we'll definitely update support in Syft!

**Comment by ankitm123** [comment]
Awesome, looking forward to it.

**Comment by kzantow** [comment]
NOTE: we need to get one more change merged into the `tools-golang` repo: https://github.com/spdx/tools-golang/pull/170

**Comment by ankitm123** [comment]
Cool, thanks! We will upgrade the version of syft used in jenkins X pipelines :)
